Sources/Network/Connection/Connection.swift

```diff
@@ -34,6 +42,22 @@ class Connection<GrpcService: ConnectionProtocol, HttpService: ConnectionProtoco
     var connectionOptionWrapper: ConnectionOptionWrapper<GrpcService, HttpService> {
         inner.accessWithoutLocking.connectionOptionWrapper
     }
+    
+    func rotateURLOnError<T>(
+        _ completion: @escaping (Result<T, ConnectionError>) -> Void
+    ) -> (Result<T, ConnectionError>) -> Void {
+        { [weak self] result in
+            switch result {
+            case .success:
+                completion(result)
+            case .failure:
+                logger.debug("ConsensusConnection - rotating config on error")
```

Comment:
fyi, you don't need to put the filename in the logging statement, its included as a default parameter this is from the implementation:

```swift
    func debug(
        _ message: @autoclosure () -> String,
        metadata: @autoclosure () -> Logging.Logger.Metadata? = nil,
        source: @autoclosure () -> String? = nil,
        sensitive: Bool = false,
        logFunction: Bool = true,
        file: String = #file,
        function: String = #function,
        line: UInt = #line
    ) {
        debug(
            Message(stringLiteral: message()),
            metadata: metadata(),
            source: source(),
            sensitive: sensitive,
            logFunction: logFunction,
            file: file,
            function: function,
            line: line)
    }
```

`#file` `#function` etc.. are special tokens that capture this info.

Reply:
ya, noticed that when reviewing the logs and meant to remove - tx for reminder.